### PR TITLE
Adding the payload in the exceptions

### PR DESCRIPTION
**Pull request by MMauro94**
Hi,
I added the payload in the custom exceptions SignatureInvalidException, BeforeValidException and ExpiredException. This may be useful if we need to inspect the payload even if the token is somehow invalid. 

Example: in a form to reset the password, if the token is expired, you can ask the user to resend the email with a new token, but to do so you need to access the user ID in the token.

### CODE REVIEW

**Comment by MMauro94** [comment]
I signed the CLA with the company name FEMA Studios S.r.l.s.

**Comment by MMauro94** [comment]
I signed it!

File: src/BeforeValidException.php [hunk 1]
```diff
@@ -3,5 +3,14 @@
 
 class BeforeValidException extends \UnexpectedValueException
 {
-
+	private $payload;
+	
+	public function __construct($payload, $message, $code = 0, $previous = null){
+		parent::__construct($message, $code, $previous);
+		$this->payload = $payload;
```
Comment (bshaffer) [inline src/BeforeValidException.php]: The styles here need to be consistent with the [coding standards of this library](http://www.php-fig.org/psr/psr-1/), i.e. 

- Four spaces for indents
- no cuddly braces for functions

**Comment by bshaffer** [comment]
For this feature, it would be better to implement this in your own implementation, for example logging the payload prior to validation, or using a try-catch.

**Pull request closed without merging**